src/Riok.Mapperly/Descriptors/MappingBodyBuilders/NewInstanceObjectMemberMappingBodyBuilder.cs

```diff
@@ -235,7 +235,7 @@ private static void BuildConstructorMapping(INewInstanceBuilderContext<IMapping>
                 continue;
             }
 
-            ctx.TargetMembers.RemoveRange(mappedTargetMemberNames);
+            ctx.TargetMembers.RemoveRange(ctx.TargetMembers.Keys.Intersect(mappedTargetMemberNames, StringComparer.OrdinalIgnoreCase));
```

Comment:
Not sure if we should adjust this to use a similar approach to `RootTargetNameCasingMapping`, wdyt?
* Create a case mapping dictionary
* Use that dictionary to resolve correct casing in `AddConstructorParameterMapping` and remove from `TargetMembers` dictionary accordingly